evm/test/RateLimit.t.sol

```diff
@@ -703,4 +720,369 @@ contract TestRateLimit is Test, IRateLimiterEvents {
             assertEq(inboundLimitParams.lastTxTimestamp, sendAgainTime);
         }
     }
+
+    // helper functions
+    function setupToken() public returns (address, address, DummyToken, uint8) {
+        address user_A = address(0x123);
+        address user_B = address(0x456);
+
+        DummyToken token = DummyToken(nttManager.token());
+
+        uint8 decimals = token.decimals();
+        assertEq(decimals, 18);
+
+        return (user_A, user_B, token, decimals);
+    }
+
+    function initializeTransceivers() public returns (ITransceiverReceiver[] memory) {
+        (DummyTransceiver e1, DummyTransceiver e2) =
+            TransceiverHelpersLib.setup_transceivers(nttManager);
+
+        ITransceiverReceiver[] memory transceivers = new ITransceiverReceiver[](2);
+        transceivers[0] = e1;
+        transceivers[1] = e2;
+
+        return transceivers;
+    }
+
+    function expectRevert(
+        address contractAddress,
+        bytes memory encodedSignature,
+        string memory expectedRevert
+    ) internal {
+        (bool success, bytes memory result) = contractAddress.call(encodedSignature);
+        require(!success, "call did not revert");
+
+        console.log("result: %s", result.length);
+        // // compare revert strings
+        bytes32 expectedRevertHash = keccak256(abi.encode(expectedRevert));
+        (bytes memory res,) = result.slice(4, result.length - 4);
+        bytes32 actualRevertHash = keccak256(abi.encodePacked(res));
+        require(expectedRevertHash == actualRevertHash, "call did not revert as expected");
+    }
+
+    // transfer tokens from user_A to user_B
+    // this consumes capacity on the outbound side
+    // send tokens from user_B to user_A
+    // this consumes capacity on the inbound side
+    // send tokens from user_A to user_B
+    // this should consume capacity on the outbound side
+    // and backfill the inbound side
+    function testFuzz_CircularFlowBackFilling(uint256 mintAmt, uint256 transferAmt) public {
+        mintAmt = bound(mintAmt, 1, type(uint256).max);
+        // enforces transferAmt <= mintAmt
+        transferAmt = bound(transferAmt, 0, mintAmt);
+
+        (address user_A, address user_B, DummyToken token, uint8 decimals) = setupToken();
+
+        // allow for amounts greater than uint64 to check if [`setOutboundLimit`] reverts
+        // on amounts greater than u64 MAX.
+        if (mintAmt.scale(decimals, 8) > type(uint64).max) {
```

Comment:
Will this kill a bunch of runs of fuzzing?

Reply:
this case should revert for sure anyway

Reply:
this case should revert in anyway